str/trtools/merge_str_runner.py

```diff
@@ -23,60 +26,100 @@
 
 
 # inputs:
-# input directory
-@click.option('--input-dir', help='gs://...')
-# input sample ID
-@click.argument('internal-wgs-ids', nargs=-1)
+
+
+# input directory 1
+@click.option('--input-dir-1', help='gs://...')
+# input directory 2
+@click.option('--input-dir-2', help='gs://...', default=None)
+# sample list 1 (CPG sample IDs separated by \n)
+@click.option('--sample-list-1', help='gs://...')
+# sample list 2 (CPG sample IDs separated by \n)
+@click.option('--sample-list-2', help='gs://...', default=None)
+# input num shards
+@click.option(
+    '--num-shards',
+    type=int,
+    default=1,
+    help=('Number of shards per sample to expect (if unsharded, choose 1)'),
+)
 @click.option(
-    '--job-storage', help='Storage of the Hail batch job eg 30G', default='50G'
+    '--job-storage', help='Storage of the Hail batch job eg 30G', default='20G'
 )
 @click.command()
 def main(
-    job_storage, input_dir, internal_wgs_ids: list[str]
+    job_storage, input_dir_1, input_dir_2, sample_list_1, sample_list_2, num_shards
 ):  # pylint: disable=missing-function-docstring
     # Initializing Batch
     b = get_batch()
+    for index in range(num_shards):
+        shard_index = index + 1  # indices start from 1
+
+        # Initialise TRTools job to run mergeSTR
+        trtools_job = b.new_job(name=f'mergeSTR shard {shard_index}')
+        trtools_job.image(TRTOOLS_IMAGE)
+        trtools_job.cpu(8)
+        trtools_job.storage(job_storage)
+        trtools_job.declare_resource_group(
+            vcf_output={
+                'vcf': '{root}.vcf',
+                'vcf.gz': '{root}.vcf.gz',
+                'vcf.gz.tbi': '{root}.vcf.gz.tbi',
+            }
+        )
+
+        # read in input file paths
+        batch_vcfs = []
+        num_samples = 0
+        with to_path(sample_list_1).open() as f_1:
+            ids_1 = [line.strip() for line in f_1]
+            for id in ids_1:
+                each_vcf = os.path.join(
+                    input_dir_1, f'{id}_eh_shard{shard_index}.reheader.vcf.gz'
+                )
+                batch_vcfs.append(
+                    b.read_input_group(
+                        **{
+                            'vcf.gz': each_vcf,
+                            'vcf.gz.tbi': f'{each_vcf}.tbi',
+                        }
+                    )['vcf.gz']
+                )
+            num_samples = num_samples + len(ids_1)
```

Comment:
I'm assuming there's no possibility of a sample existing in both locations?

Reply:
not with my current understanding of this dataset. The merge step _should_ throw an error if there were duplicate samples because we didn't specify the flag that allows dups.